pkg/apis/elasticsearch/v1/elasticsearch_types_test.go

```diff
@@ -314,3 +316,88 @@ func TestElasticsearch_DisabledPredicates(t *testing.T) {
 		})
 	}
 }
+
+// Test_AssociationConfs tests that if something resets the AssocConfs map, then AssociationConf() reinitializes
+// the map from the annotation.
```

Comment:
Not sure about `resets`, I would use `clear`, but it might come from my [Java background](https://docs.oracle.com/javase/7/docs/api/java/util/HashMap.html#clear()).
```suggestion
// Test_AssociationConfs tests that if the association configuration map in an associated object is cleared, then AssociationConf() is rebuilt from the annotation.
```